packages/core/src/blockchain/storage-layer.ts

```diff
@@ -249,7 +326,7 @@ export class StorageLayer implements StorageLayerProvider {
       const next = await this.findNextKey(prefix, startKey, undefined)
       if (!next) break
       startKey = next
-      if ((await this.get(next, false)) === StorageValueKind.Deleted) continue
+      if (this.#isDeleted(next)) continue
```

Comment:
This was causing that when someone called `getKeysPaged`, it would also fetch the values of those keys, only to check if it was deleted in the local cache.

So I have refactored it to just check for deletion without calling `get`